### PR TITLE
simdutf: init at 4.0.5

### PR DESCRIPTION
**Pull request by wineee**
## Description of changes

https://github.com/simdutf/simdutf

Unicode routines (UTF8, UTF16, UTF32): billions of characters per second using SSE2, AVX2, NEON, AVX-512. Part of Node.js and Bun.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/si/simdutf/package.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-HNTVo/uB7UTCy5VVdmf6vka9T+htra7Vk7NF4hByGP4=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  meta = with lib; {
+    description = "Unicode routines (UTF8, UTF16, UTF32): billions of characters per second";
```
Comment (AndersonTorres) [inline pkgs/by-name/si/simdutf/package.nix:23]:
```suggestion
    description = "Unicode validation and transcoding at billions of characters per second";
```

A better description imnsho.

**Review by AndersonTorres — APPROVED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Review by katexochen — APPROVED** [review]
Result of `nixpkgs-review pr 270395` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>simdutf</li>
  </ul>
</details>

File: pkgs/by-name/si/simdutf/package.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  pname = "simdutf";
+  version = "4.0.5";
+
+  src = fetchFromGitHub {
+    owner = "simdutf";
+    repo = "simdutf";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-HNTVo/uB7UTCy5VVdmf6vka9T+htra7Vk7NF4hByGP4=";
+  };
+
```
Comment (emilytrau) [inline pkgs/by-name/si/simdutf/package.nix:17]:
```suggestion
  postPatch = ''
    substituteInPlace tools/CMakeLists.txt --replace "-Wl,--gc-sections" ""
  '';
```

**Review by emilytrau — CHANGES_REQUESTED** [review]
Fails to build on darwin

```
[ 14%] Linking CXX executable sutf
ld: unknown option: --gc-sections
clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [tools/CMakeFiles/sutf.dir/build.make:98: tools/sutf] Error 1
make[1]: *** [CMakeFiles/Makefile2:2304: tools/CMakeFiles/sutf.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 14%] Linking CXX static library libsimdutf_tests_helpers.a
[ 14%] Built target simdutf_tests_helpers
make: *** [Makefile:136: all] Error 2
error: boost::bad_format_string: format-string is ill-formed
```

**Review by emilytrau — APPROVED** [review]

**Review by wegank — CHANGES_REQUESTED** [review]
```
[100%] Linking CXX executable sutf
Undefined symbols for architecture arm64:
  "_iconv", referenced from:
      CommandLine::iconv_fallback(__sFILE*) in sutf.cpp.o
  "_iconv_close", referenced from:
      CommandLine::iconv_fallback(__sFILE*) in sutf.cpp.o
  "_iconv_open", referenced from:
      CommandLine::iconv_fallback(__sFILE*) in sutf.cpp.o
ld: symbol(s) not found for architecture arm64
clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
```

`libiconv` needs to be added to `buildInputs` on aarch64-darwin. LGTM otherwise.

File: pkgs/by-name/si/simdutf/package.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  # Fix build on darwin
+  postPatch = ''
+    substituteInPlace tools/CMakeLists.txt --replace "-Wl,--gc-sections" ""
+  '';
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  buildInputs = [
```
Comment (wegank) [inline pkgs/by-name/si/simdutf/package.nix:28]:
```suggestion
  buildInputs = lib.optionals stdenv.isDarwin [
```

**Comment by wineee** [comment]
@ofborg build simdutf

**Review by wegank — APPROVED** [review]

File: pkgs/by-name/si/simdutf/package.nix (hunk 3 above)
Comment (wineee) [reply pkgs/by-name/si/simdutf/package.nix:28]: libiconv is also used in linux, but it stemming from the C library, so add it to buildInputs is not necessary but should acceptable

```
/nix/store/w9mn7nhrd36xrsa8pn141ad0azkkhbxa-glibc-iconv-2.38
└── include
   └── iconv.h -> /nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/iconv.h
```

**Comment by emilytrau** [comment]
Can't see why ofborg is complaining on just `aarch64-darwin` but looks good to me 👍

**Pull request merged**